narwhals/dtypes.py

```diff
@@ -426,10 +452,34 @@ def __init__(
             time_zone = str(time_zone)
 
         self.time_unit: TimeUnit = time_unit
+        """Unit of time."""
         self.time_zone: str | None = time_zone
+        """Time zone string, as defined in zoneinfo.
+
+        Notes:
+            To see valid strings run `import zoneinfo; zoneinfo.available_timezones()` for a full list.
+        """
+
+    def __eq__(self, other: DType | type[DType]) -> bool:  # type: ignore[override]
+        """Check if this Datetime is equivalent to another DType.
 
-    def __eq__(self, other: object) -> bool:
-        # allow comparing object instances to class
+        Examples:
+            >>> import narwhals as nw
+            >>> nw.Datetime("s") == nw.Datetime("s")
+            True
+            >>> nw.Datetime("us") == nw.Datetime()
```

Comment:
self-suggest (https://github.com/narwhals-dev/narwhals/pull/2999/files/cccc9ca03e982c56e2518fe53072c5c4bd29c829#r2280627830)
```suggestion
            >>> nw.Datetime() == nw.Datetime("us")
```